tests/mesh/mesh_smoother_test.C

```diff
@@ -269,15 +278,25 @@ public:
   {
     LOG_UNIT_TEST;
 
-    const auto dim = ReferenceElem::get(type).dim();
+    // Get mesh dimension, determine whether type is triangular
+    const auto * ref_elem = &(ReferenceElem::get(type));
+    const auto dim = ref_elem->dim();
+    const bool type_is_tri = dynamic_cast<const Tri*>(ref_elem);
```

Comment:
Not necessary for this PR, but this feels so clunky that we should probably add a helper function that does this by looking only at an `ElemType` enum. Not sure what the best place would be, but maybe a helper function like `enum_to_string()` would probably make sense. It would need to be updated as new Tri types are added, but we have a few things like that already so I wouldn't mind one more.